R/utils.R

```diff
@@ -950,20 +950,14 @@ columnToRowData <- function(data) {
 #' }
 #' @export
 safeError <- function(error, errorClass = character(0)) {
-  if (inherits(error, "error")) {
-    cond <- structure(
-      list(message = error$message),
-      class = c("shiny.custom.error", errorClass, class(error))
-    )
-  } else if (inherits(error, "character")) {
-    cond <- structure(
-      list(message = error),
-      class = c("shiny.custom.error", errorClass, "error", "condition")
-    )
-  } else {
+  if (inherits(error, "character")) {
+    error <- simpleError(error)
+  }
+  if (!inherits(error, "error")) {
     stop("The class of the `error` parameter must be either 'error' or 'character'")
   }
-  return(cond)
+  class(error) <- c("shiny.custom.error", errorClass, class(error))
```

Comment:
Hmmm... Initially, I just took errorClass from `stopWithCondition` that has an argument just like that. I then thought it was worth keeping around to allow app authors to style their apps however they want (yes, we do give them the "shiny.custom.error" class, but maybe one class is not enough). I doubt this will be used a lot, but it seemed unobtrusive enough that it was worth keeping it around to give app authors more control over their errors.